src/main/java/org/opensearch/searchrelevance/transport/experiment/GetExperimentTransportAction.java

```diff
@@ -44,11 +42,8 @@ public GetExperimentTransportAction(
 
     @Override
     protected void doExecute(Task task, OpenSearchDocRequest request, ActionListener<SearchResponse> listener) {
-        // Validate cluster health first
-        if (!clusterService.state().routingTable().hasIndex(EXPERIMENT.getIndexName())) {
-            listener.onFailure(new ResourceNotFoundException("Index [" + EXPERIMENT.getIndexName() + "] not found"));
-            return;
-        }
+        StepListener<Void> createIndexStep = new StepListener<>();
+        experimentDao.createIndexIfAbsent(createIndexStep);
```

Comment:
createIndexIfAbsent() method will make async call, so it's possible that flow reaches your next calls in this (doExecute) method before the index get created. You need to run next actions when createIndexIfAbsent is 1) completed and 2) done with success. And if it failed for whatever reason you need to interrupt the process. 

Take as example this snippet from put experiment action https://github.com/opensearch-project/search-relevance/blob/main/src/main/java/org/opensearch/searchrelevance/transport/experiment/PutExperimentTransportAction.java#L103-L126